### PR TITLE
Meter system enhancement

### LINKED ISSUE #6356: Meter system enhancement.
MAL is widely used in different meter/metrics receivers today, we are facing a challenge, which is, when we share the same metrics(name and data type) over different MAL script definition, the `MeterSystem#create` fails.

We need to make `MeterSystem#create` to return `true` w/o failing when `metricsName`/`functionName`/`type` are same.

### PR DESCRIPTION
**Pull request by mrproliu**
### Fix #6356 
- [x] Add a unit test to verify that the fix works.
- [x] Support create metric with same `metricName` / `function` / `scope`.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java [hunk 1]
```diff
@@ -187,6 +187,22 @@ public MeterSystem(final ModuleManager manager) {
...
         final String className = formatName(metricsName);
+
+        /**
+         * Check is already define the metric
+         */
+        try {
+            CtClass alreadyMetric = classPool.get(METER_CLASS_PACKAGE + className);
+            if (alreadyMetric.getSuperclass() != parentClass || type != meterPrototypes.get(metricsName).getScopeType()) {
+                throw new IllegalArgumentException("Already define metric " + metricsName
+                                                       + ", but calculate function or scope type is not same.");
```
Comment (kezhenxu94) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:198]:
```suggestion
                throw new IllegalArgumentException("Already defined metrics class " + metricsName
                                                       + ", but calculate function or scope type is different.");
```

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java [hunk 2]
```diff
@@ -187,6 +187,21 @@ public MeterSystem(final ModuleManager manager) {
             throw new IllegalArgumentException("Function " + functionName + " can't be found by javaassist.");
         }
         final String className = formatName(metricsName);
+
+        /**
+         * Check whether the metrics class is already defined or not
+         */
+        try {
+            CtClass alreadyMetric = classPool.get(METER_CLASS_PACKAGE + className);
+            if (alreadyMetric.getSuperclass() != parentClass || type != meterPrototypes.get(metricsName).getScopeType()) {
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:196]: `!=` -> `equals`?

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java [hunk 3]
```diff
@@ -187,6 +187,21 @@ public MeterSystem(final ModuleManager manager) {
             throw new IllegalArgumentException("Function " + functionName + " can't be found by javaassist.");
         }
         final String className = formatName(metricsName);
+
+        /**
+         * Check whether the metrics class is already defined or not
+         */
+        try {
+            CtClass alreadyMetric = classPool.get(METER_CLASS_PACKAGE + className);
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:195]:
```suggestion
            CtClass existingMetric = classPool.get(METER_CLASS_PACKAGE + className);
```

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java [hunk 4]
```diff
@@ -187,6 +187,21 @@ public MeterSystem(final ModuleManager manager) {
...
+            if (alreadyMetric.getSuperclass() != parentClass || type != meterPrototypes.get(metricsName).getScopeType()) {
+                throw new IllegalArgumentException("Already defined metrics class " + metricsName
+                                                       + ", but calculate function or scope type is different.");
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:198]:
```suggestion
                                                       + ", but calculate function or/are scope type is/are different.");
```

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java [hunk 5]
```diff
@@ -187,6 +187,21 @@ public MeterSystem(final ModuleManager manager) {
...
+            }
+            log.debug("Metric {} is already define, so skip the metric creation.", metricsName);
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:200]:
```suggestion
            log.error("Metric {} is already define, so skip the metric creation.", metricsName);
```

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java (hunk 4 above)
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:198]: log -> Metric class + metricsName + has been defined, but calculate function or/are scope type is/are different.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java (hunk 2 above)
Comment (mrproliu) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:196]: Their data type is `Class`, I think we don't need to the use `equals` method?

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:196]: Make sense, they share the same classPool, so same class loader. Technically, yes.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java (hunk 5 above)
Comment (mrproliu) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:200]: Such as the `OTEL` and `Zabbix` plugin, use the same metric name, I think it's a normal log. Why need to upgrade to the error level?

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:200]: I see your point. Keep it in INFO at least, this is important.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java [hunk 6]
```diff
@@ -187,6 +185,20 @@ public MeterSystem(final ModuleManager manager) {
...
+
+        /**
+         * Check whether the metrics class is already defined or not
+         */
+        try {
+            CtClass existingMetric = classPool.get(METER_CLASS_PACKAGE + className);
+            if (existingMetric.getSuperclass() != parentClass || type != meterPrototypes.get(metricsName).getScopeType()) {
+                throw new IllegalArgumentException(metricsName + " has been defined, but calculate function or/are scope type is/are different.");
+            }
+            log.info("Metric {} is already define, so skip the metric creation.", metricsName);
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/meter/MeterSystem.java:197]:
```suggestion
            log.info("Metric {} is already defined, so skip the metric creation.", metricsName);
```

**Review by wu-sheng — APPROVED** [review]
Once tests passed, good to me.

**Pull request merged**